### PR TITLE
Relatedproducts

### PR DESCRIPTION
**Pull request by ahusu**
Made A bunch of changes.  Modal is now rendering the features correctly, (still cant get price to be centered =(...) and the RP cards are the same size.  Having some trouble containing the card image correctly

### CODE REVIEW

File: src/components/relatedProducts/ComparisonModal.jsx [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
   const dispatch = useDispatch();
   const productId = useSelector((state) => state.productId);
-  const comparisonId = useSelector(state=>state.comparisonId);
+  const comparisonId = useSelector(state => state.comparisonId);
   const [baseDetails, setBaseDetails] = useState({});
   const [comparedDetails, setComparedDetails] = useState({});
-  let features = new Set([]);
+  const [baseStars, setBaseStars] = useState(0);
+  const [comparedStars, setComparedStars] = useState(0);
+  const [features, setFeatures] = useState([])
```
Comment (carolinerobbins) [inline src/components/relatedProducts/ComparisonModal.jsx:16]: just confirming - features is the combination (set) of features from both products you are comparing?

File: src/components/relatedProducts/ComparisonModal.jsx [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
+        console.log(feat)
+        bFeatures = {...bFeatures, ...feat }
+      });
+      compared.features.forEach((featureObj) => {
+        featuresSet.add(featureObj.feature)
+        let feat = {};
+        feat[featureObj.feature] = featureObj.value
+        cFeatures = {...cFeatures, ...feat }
+      });
+      const set = [...featuresSet]
```
Comment (carolinerobbins) [inline src/components/relatedProducts/ComparisonModal.jsx:44]: QQ - why do you need to use a spread operator and reassign featuresSet before updating state?

**Review by carolinerobbins — APPROVED** [review]
LGTM - added 2 questions on the modal (more for me to understand vs. changes needed)

File: src/components/relatedProducts/ComparisonModal.jsx (hunk 1 above)
Comment (ahusu) [reply src/components/relatedProducts/ComparisonModal.jsx:16]: Yep! 👍👍

File: src/components/relatedProducts/ComparisonModal.jsx (hunk 2 above)
Comment (ahusu) [reply src/components/relatedProducts/ComparisonModal.jsx:44]: Sets don't have all of the methods arrays do.  In this case I'm just passing into an array to give it standard array methods.

**Pull request merged**